### PR TITLE
Make sure the UriTemplate can parse two or more adjacent variables separated by dot

### PR DESCRIPTION
**Pull request by corleonis**
So this is kind of dirty solution. It works but probably not structured the best way. Since I am new to that project I feel I am getting lost in the code. This is to spark a bit of discussion and try to move things along as I really need that feature for our project, it's critical.

### CODE REVIEW

File: src/Rize/UriTemplate/Parser.php [hunk 1]
```diff
@@ -21,7 +23,20 @@ public function parse($template)
         $nodes   = array();
 
         foreach($parts as $part) {
-            $nodes[] = $this->createNode($part);
+            $node = $this->createNode($part);
+
+            // if current node has dot separator that requires a forward lookup
+            // for the previous node iff previous node's operator is UnNamed
+            if ($node instanceof Expression && $node->getOperator()->id == '.') {
```
Comment (rezigned) [inline src/Rize/UriTemplate/Parser.php]: Maybe we should use strict comparison `===` instead.

File: test/Rize/Uri/Node/ParserTest.php [hunk 2]
```diff
@@ -85,4 +86,45 @@ public function testParseTemplate()
 
         $this->assertEquals($expected, $actual);
     }
-}
\ No newline at end of file
+
+    public function testParseTemplateWithLiteral()
+    {
+        // will pass
+        $uri = new UriTemplate('http://www.example.com/v1/company/', []);
```
Comment (rezigned) [inline test/Rize/Uri/Node/ParserTest.php]: Seems like `[]` makes tests in travis-ci failed in php 5.3 (even though it's no longer supported). Maybe you can remove this or change to `array()`)

Comment (corleonis) [inline test/Rize/Uri/Node/ParserTest.php]: ah good spot will fix that, also there is another issue with composer running on PHP 5.3.3 on Travis but I believe that is a generic problem with their container:.

**Comment by corleonis** [comment]
There is problem with composer on PHP5.3.3 on Travis here is the issue on GH: https://github.com/composer/composer/issues/5388

**Comment by rezigned** [comment]
Could you do me a favor and remove `5.3.3` from this file https://github.com/rize/UriTemplate/blob/master/.travis.yml ? Thank you.

**Comment by rezigned** [comment]
Done. Added `0.3.1` tag. Big Thanks!

**Pull request merged**